shared/src/main/scala/zeno/examples/MultiPaxos.proto

```diff
@@ -0,0 +1,108 @@
+syntax = "proto2";
+
+import "scalapb/scalapb.proto";
+
+option (scalapb.options) = {
+  package_name: "zeno.examples"
+  flat_package: true
+};
+
+// Incoming client requests containing command they want to get chosen
+message ClientRequest {
+  required string command = 1;
+}
+
+// Messages sent between replicas and leaders.
+message ProposeToLeader {
+  required int32 slot = 1;
+  required string command = 2;
+}
+
+// Messages sent to decide command for a slot
+message Decision {
+  required int32 slot = 1;
+  required string command = 2;
+}
+
+// Struct holding the ballot, slot, and command for a given proposal
+message ProposedValue {
+  required double ballot = 1;
+  required int32 slot = 2;
+  required string command = 3;
+}
+
+// Indicates that majority of acceptors accept ballot, also contains what acceptors accepted before
+message Adopted {
+  required double ballot = 1;
+  repeated ProposedValue proposals = 2;
+}
+
+// Indicates acceptor has accepted higher ballot, need to abandon this ballot
+message Preempted {
```

Comment:
I would leave this here, as it's a super important optimization. But on your first go at an implementation, to keep things simple, I would leave this unimplemented, and then implement it later after you've confirmed that the code is working properly.